subsys/bluetooth/mesh/net.c

```diff
@@ -89,6 +96,9 @@ struct bt_mesh_net bt_mesh = {
 	.local_queue = SYS_SLIST_STATIC_INIT(&bt_mesh.local_queue),
 };
 
+/* Time when last IV Index Recovery was done. */
+static int64_t ivi_recovery;
```

Comment:
Since this value is only set when performing an IV update, and that already runs on a 96+96H time limit, would it be enough to just keep a flag here? e.g. `IV_INDEX_WAS_RECOVERED`, and reset that to false if we change the IV index in a normal way?

Reply:
I really like this idea, but is not the IV update timer configurable. So in real it can be longer, but maybe that does not matter.

Reply:
Yeah, like the 96H IV update delay, we only need the minimum time to pass

Reply:
This will again create an issue because the iv_duration timer is intermittently stored in the flash. 

Just to recap what we discussed in the past. The device needs to have the ability to recover the IV index if it is offline when the rest of the network has progressed to the last known IV + 1. There is also a spec requirement that IV recovery can happen only once every 192 hours. When the device is powered off, it loses the track of time in the current stack implementation (although for overall IV procedure there is a timer which is intermittently stored in flash. But under a certain pattern of off-on cycles, even this timer will lag the rest of the network eventually causing the device to fall out of the IV update cycle). With that assumption, the change in this PR allows the device to recover IV exactly once every 192 hours when the device is active without rebooting. This meets the spec constraint for IV recovery. If the device is offline for a brief period of time, the recovery procedure will anyway not trigger because the IV known by the device would match with the IV of the network.

In short, the ivi_recovery should stay volatile and remain separate from iv.iv_duration.

Reply:
ivi_was_recovered is not stored and will be reset on reboot. So the functionality is still the same.